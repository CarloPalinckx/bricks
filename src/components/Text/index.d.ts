import { SFC } from 'react';
import { StyledType } from '../../utility/styled';
import SeverityType from '../../types/SeverityType';
declare type PropsType = StyledType & {
    compact?: boolean;
    descriptive?: boolean;
    strong?: boolean;
    inline?: boolean;
    variant?: 'small' | 'base' | 'large';
    textAlign?: 'left' | 'right' | 'center' | 'justify';
    severity?: SeverityType;
};
declare const Text: SFC<PropsType>;
export default Text;
export { PropsType };
