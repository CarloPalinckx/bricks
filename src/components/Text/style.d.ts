import _R from 'react';
import { StyledComponentClass as _S } from 'styled-components';
import _T from '../../types/ThemeType';
import SeverityType from '../../types/SeverityType';
declare type TextVariantStyleType = {
    fontSize: string;
};
declare type SeverityStyleType = {
    color: string;
};
declare type TextThemeType = {
    default: {
        color: string;
        fontFamily: string;
        fontSize: string;
        fontWeight: string;
    };
    descriptive: {
        color: string;
    };
    strong: {
        fontWeight: string;
    };
    severity: {
        success: SeverityStyleType;
        warning: SeverityStyleType;
        error: SeverityStyleType;
        info: SeverityStyleType;
    };
    small: TextVariantStyleType;
    base: TextVariantStyleType;
    large: TextVariantStyleType;
};
declare type TextPropsType = {
    compact?: boolean;
    descriptive?: boolean;
    strong?: boolean;
    variant?: 'small' | 'base' | 'large';
    textAlign?: 'left' | 'right' | 'center' | 'justify';
    severity?: SeverityType | '';
};
declare const StyledParagraph: _S<TextPropsType & _R.HTMLProps<HTMLParagraphElement>, _T, TextPropsType & _R.HTMLProps<HTMLParagraphElement>>;
declare const StyledSpan: _S<_R.DetailedHTMLProps<_R.HTMLAttributes<HTMLSpanElement>, HTMLSpanElement>, _T, TextPropsType & _R.HTMLProps<HTMLParagraphElement>>;
export { StyledSpan, StyledParagraph, TextThemeType, TextVariantStyleType, SeverityStyleType };
