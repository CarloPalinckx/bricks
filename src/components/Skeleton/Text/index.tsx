import React, { Fragment, FunctionComponent } from 'react';
import StyledTextSkeleton from './style';

type TextPropsType = {
    lines: number;
    baseWidth?: number;
};

const Text: FunctionComponent<TextPropsType> = ({ lines, baseWidth }): JSX.Element => {
    const generatedLines: Array<JSX.Element> = [];

    if (baseWidth !== undefined) baseWidth = baseWidth < 6 ? 6 : baseWidth;

    for (let i = 0; i < lines; i++) {
        generatedLines.push(
            <Fragment key={i}>
                <StyledTextSkeleton aria-hidden baseWidth={baseWidth}>
                    _
                </StyledTextSkeleton>
                {i !== lines - 1 && <br />}
            </Fragment>,
        );
    }

    return <>{generatedLines}</>;
};

export default Text;
export { TextPropsType };
