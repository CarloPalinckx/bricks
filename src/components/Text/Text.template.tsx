import React, { StatelessComponent } from 'react';
import { StyledType } from '../../utility/styled';

type PropsType = StyledType & {
    strong?:boolean;
    descriptive?:boolean;
};

const Text:StatelessComponent<PropsType> = (props):JSX.Element => (
    <p className={props.className}>{props.children}</p>
);

export default Text;
export {
    PropsType,
};
