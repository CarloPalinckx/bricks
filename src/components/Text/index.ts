export { default, TextThemeType } from './Text.style';
