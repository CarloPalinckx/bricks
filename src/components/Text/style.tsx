import _R from 'react';
import { StyledComponentClass as _S } from 'styled-components';
import _T from '../../types/ThemeType';
import styled, { withProps } from '../../utility/styled';
import SeverityType from '../../types/SeverityType';

type TextVariantStyleType = {
    fontSize: string;
};

type SeverityStyleType = {
    color: string;
};

type TextThemeType = {
    default: {
        color: string;
        fontFamily: string;
        fontSize: string;
        fontWeight: string;
    };
    descriptive: {
        color: string;
    };
    strong: {
        fontWeight: string;
    };
    severity: {
        success: SeverityStyleType;
        warning: SeverityStyleType;
        error: SeverityStyleType;
        info: SeverityStyleType;
    };
    small: TextVariantStyleType;
    base: TextVariantStyleType;
    large: TextVariantStyleType;
};

type TextPropsType = {
    compact?: boolean;
    descriptive?: boolean;
    strong?: boolean;
    variant?: 'small' | 'base' | 'large';
    textAlign?: 'left' | 'right' | 'center' | 'justify';
    severity?: SeverityType | '';
};

const StyledParagraph = withProps<TextPropsType, HTMLParagraphElement>(styled.p)`
    color: ${({ descriptive, severity, theme }): string => {
        if (descriptive !== undefined && descriptive) {
            return theme.Text.descriptive.color;
        }

        if (severity !== undefined && severity !== '') {
            return theme.Text.severity[severity].color;
        }

        return theme.Text.default.color;
    }}
    font-family: ${({ theme }): string => theme.Text.default.fontFamily};
    font-size: ${({ variant, theme }): string => (variant ? theme.Text[variant].fontSize : theme.Text.base.fontSize)};
    font-weight: ${({ strong, theme }): string =>
        strong ? theme.Text.strong.fontWeight : theme.Text.default.fontWeight};
    line-height: ${({ compact }): string => (compact ? '1.25' : '1.5')};
    text-align: ${({ textAlign }): string => (textAlign ? textAlign : '')};
    margin: 0;
`;

const StyledSpan = StyledParagraph.withComponent('span');

export { StyledSpan, StyledParagraph, TextThemeType, TextVariantStyleType, SeverityStyleType };
