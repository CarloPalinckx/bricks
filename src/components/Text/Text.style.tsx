import { StyledComponentClass as _S } from 'styled-components';
import styled from '../../utility/styled';
import Text from './Text.template';

type TextThemeType = {
    default:{
        color:string;
        fontFamily:string;
        fontSize:string;
        fontWeight:string;
    };
    descriptive:{
        color:string;
    };
    strong:{
        fontWeight:string;
    };
};

const StyledText = styled(Text)`
    color: ${({ descriptive, theme }):string => descriptive ? theme.Text.descriptive.color : theme.Text.default.color};
    font-family: ${({ theme }):string => theme.Text.default.fontFamily};
    font-size: ${({ theme }):string => theme.Text.default.fontSize};
    font-weight: ${({ strong, theme }):string => strong ? theme.Text.strong.fontWeight : theme.Text.default.fontWeight};
    line-height: 1.5;
    margin: 0;
`;

export default StyledText;
export {
    TextThemeType,
};
