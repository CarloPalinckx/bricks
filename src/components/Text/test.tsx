import toJson from 'enzyme-to-json';
import React from 'react';
import Text from '.';
import { shallowWithTheme } from '../../utility/styled';

describe('Text', () => {
    it('should render text with default styling', () => {
        const component = shallowWithTheme(<Text>Default text</Text>);

        expect(toJson(component.dive())).toMatchSnapshot();
    });

    it('should render text with strong styling', () => {
        const component = shallowWithTheme(<Text strong>Strong text</Text>);

        /* tslint:disable */
        (expect(toJson(component.dive())) as any).toHaveStyleRule('font-weight', '700');
        /* tslint:enable */
    });

    it('should render text with descriptive styling', () => {
        const component = shallowWithTheme(<Text descriptive>Descriptive text</Text>);

        /* tslint:disable */
        (expect(toJson(component.dive())) as any).toHaveStyleRule('color', '#a6aab3');
        /* tslint:enable */
    });

    it('should render text with compact styling', () => {
        const component = shallowWithTheme(<Text compact>Descriptive text</Text>);

        /* tslint:disable */
        (expect(toJson(component.dive())) as any).toHaveStyleRule('line-height', '1.25');
        /* tslint:enable */
    });

    it('should render a span when inline is set to true', () => {
        const component = shallowWithTheme(<Text inline>Descriptive text</Text>);

        expect(component.dive().find('span').length).toBe(1);
    });
});
