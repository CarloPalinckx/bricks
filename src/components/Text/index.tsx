import React, { StatelessComponent } from 'react';
import { StyledType } from '../../utility/styled';
import { StyledSpan, StyledParagraph } from './style';
import SeverityType from '../../types/SeverityType';

type PropsType = StyledType & {
    compact?: boolean;
    descriptive?: boolean;
    strong?: boolean;
    inline?: boolean;
    variant?: 'small' | 'base' | 'large';
    textAlign?: 'left' | 'right' | 'center' | 'justify';
    severity?: SeverityType;
};

const Text: StatelessComponent<PropsType> = (props): JSX.Element => {
    const Element = props.inline !== undefined && props.inline ? StyledSpan : StyledParagraph;

    return (
        <Element
            variant={props.variant}
            descriptive={props.descriptive}
            strong={props.strong}
            compact={props.compact}
            textAlign={props.textAlign}
            severity={props.severity}
        >
            {props.children}
        </Element>
    );
};

export default Text;
export { PropsType };
