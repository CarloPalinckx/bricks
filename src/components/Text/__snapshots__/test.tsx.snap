// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render text with default styling 1`] = `
.c0 {
  color: #333740;
  font-family: Source Sans Pro,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1.5;
  margin: 0;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

<p
  className="c0"
>
  Default text
</p>
`;
